provider/cmd/crd2pulumi/gen/nodejs.go

```diff
@@ -34,34 +32,14 @@ const metaFile = `import * as k8s from "@pulumi/kubernetes";
 export type ObjectMeta = k8s.types.input.meta.v1.ObjectMeta;
 `
 
-var alphanumericRegex = regexp.MustCompile("[^a-zA-Z0-9]+")
-
-// removes all non-alphanumeric characters
-func removeNonAlphanumeric(input string) string {
-	return alphanumericRegex.ReplaceAllString(input, "")
-}
-
-// un-capitalizes the first character of a string
-func toLowerFirst(input string) string {
-	if input == "" {
-		return ""
-	}
-	return string(unicode.ToLower(rune(input[0]))) + input[1:]
-}
-
 func (pg *PackageGenerator) genNodeJS(types map[string]pschema.ObjectTypeSpec, baseRefs []string) (map[string]*bytes.Buffer, error) {
```

Comment:
FYI, I am running into the following assert when running this locally:

```
$ go run . nodejs resourcedefinition.yaml out/nodejs
panic: fatal: An assertion has failed

goroutine 1 [running]:
github.com/pulumi/pulumi/sdk/v2/go/common/util/contract.failfast(...)
	/Users/justin/go/pkg/mod/github.com/pulumi/pulumi/sdk/v2@v2.9.2/go/common/util/contract/failfast.go:23
github.com/pulumi/pulumi/sdk/v2/go/common/util/contract.Assert(...)
	/Users/justin/go/pkg/mod/github.com/pulumi/pulumi/sdk/v2@v2.9.2/go/common/util/contract/assert.go:26
github.com/pulumi/pulumi-kubernetes/provider/v2/cmd/crd2pulumi/gen.(*PackageGenerator).genNodeJS(0xc000121c90, 0xc000146960, 0xc000612930, 0x1, 0x1, 0x1, 0x1, 0xc0006128e0)
	/Users/justin/go/src/github.com/pulumi/pulumi-kubernetes/provider/cmd/crd2pulumi/gen/nodejs.go:69 +0xbb0
github.com/pulumi/pulumi-kubernetes/provider/v2/cmd/crd2pulumi/gen.(*PackageGenerator).generateFiles(0xc000121c90, 0x6, 0x7ffeefbff9ae, 0x17)
	/Users/justin/go/src/github.com/pulumi/pulumi-kubernetes/provider/cmd/crd2pulumi/gen/generate.go:168 +0x25c
github.com/pulumi/pulumi-kubernetes/provider/v2/cmd/crd2pulumi/gen.Generate(0x7ffeefbff9a7, 0x6, 0x7ffeefbff9ae, 0x17, 0x7ffeefbff9c6, 0xa, 0xc000121d00, 0x1c03ce0, 0xc0007050a0)
	/Users/justin/go/src/github.com/pulumi/pulumi-kubernetes/provider/cmd/crd2pulumi/gen/generate.go:59 +0x159
github.com/pulumi/pulumi-kubernetes/provider/v2/cmd/crd2pulumi/cmd.glob..func2(0x26b7ae0, 0xc000133a70, 0x3, 0x3)
	/Users/justin/go/src/github.com/pulumi/pulumi-kubernetes/provider/cmd/crd2pulumi/cmd/root.go:54 +0x105
github.com/spf13/cobra.(*Command).execute(0x26b7ae0, 0xc0001ac010, 0x3, 0x3, 0x26b7ae0, 0xc0001ac010)
	/Users/justin/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0x26b7ae0, 0x0, 0x0, 0xc000180058)
	/Users/justin/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/justin/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/pulumi/pulumi-kubernetes/provider/v2/cmd/crd2pulumi/cmd.Execute(...)
	/Users/justin/go/src/github.com/pulumi/pulumi-kubernetes/provider/cmd/crd2pulumi/cmd/root.go:68
main.main()
	/Users/justin/go/src/github.com/pulumi/pulumi-kubernetes/provider/cmd/crd2pulumi/main.go:22 +0x2e
exit status 2

```

Though, I think you've force pushed some changes since I've run this, so not sure if it's still an issue.